modules/overviews/app/components/project_custom_fields/sections/show_component.html.erb

```diff
@@ -1,37 +1,32 @@
 <%=
   component_wrapper do
-    flex_layout(border: :bottom, pb: 2, classes: 'op-project-custom-field-section-container', data: {
+    render(Primer::OpenProject::SidePanel::Section.new(
+      classes: 'op-project-custom-field-section-container',
       test_selector: "project-custom-field-section-#{@project_custom_field_section.id}"
-    }) do |details_container|
-      details_container.with_row(mb: 2) do
-        flex_layout(align_items: :center, justify_content: :space_between) do |heading|
-          heading.with_column(flex: 1) do
-            render(Primer::Beta::Heading.new(tag: :h4)) { @project_custom_field_section.name }
-          end
+    )) do |section|
+      section.with_title { @project_custom_field_section.name }
 
-          heading.with_column do
-            render(Primer::Beta::IconButton.new(
-              icon: :pencil,
-              tag: :a,
-              href: project_custom_field_section_dialog_path(project_id: @project.id, section_id: @project_custom_field_section.id),
-              data: {
-                controller: 'async-dialog'
-              },
-              scheme: :invisible,
-              button_icon: :pencil,
-              test_selector: "project-custom-field-section-edit-button",
-              aria: { label: I18n.t(:label_edit) },
-            ))
-          end if allowed_to_edit?
-        end
+      if allowed_to_edit?
+        section.with_action_icon(
+          icon: :pencil,
+          tag: :a,
+          href: project_custom_field_section_dialog_path(project_id: @project.id, section_id: @project_custom_field_section.id),
+          data: {
+            controller: 'async-dialog'
```

Comment:
Here there is again the same issue with the async dialog as in the other PR: When submitting an erroneous form, there is no error message shown, but just the dialog.